src/odemis/driver/technolution.py

```diff
@@ -368,69 +435,92 @@ def _acquire(self):
 
                 tnext = time.time() + 0.2  # max 5 Hz
 
-        except Exception as e:
-            if command == "terminate":
-                raise Exception("Terminated acquisition")
+        except TerminationRequested:
+            logging.info("Terminating acquisition")
 
-            elif command is not None:
-                logging.error("last message was not executed, should have perfomed action: '%s'\n"
-                              "received exepction: \n %s" % (command, e))
+        except Exception:
+            if command is not None:
+                logging.exception("Last message was not executed, should have perfomed action: '%s'\n"
+                                  "Reinitialize and restart the acquisition" % command)
         finally:
-            self.parent.ASMAPICall(self._server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+            self.parent.ASM_API_Post_Call("/scan/finish_mega_field", 204)
             logging.debug("Acquisition thread ended")
 
     def start_acquisition(self):
+        """
+        Put a the command 'start' mega field scan on the queue with the appropriate MegaFieldMeta Model of the mega
+        field image to be scannend. All subsequent calls to scan_field will use a part of this meta data to store the image
+        data until the stop command is executed.
+        """
+        if not self._acq_thread or not self._acq_thread.is_alive():
+            logging.info('Starting acquisition thread and clearing remainder of the old queue')
+
+            # Clear the queue
+            while True:
+                try:
+                    self.acq_queue.get(block=False)
+                except queue.Empty:
+                    break
+
+            self._acq_thread = threading.Thread(target=self._acquire,
+                                                name="acquisition thread")
+            self._acq_thread.deamon = False
+            self._acq_thread.start()
+
         megafield_metadata = self._assemble_megafield_metadata()
         self.acq_queue.put(("start", megafield_metadata))
 
     def get_next_field(self, field_num):
         '''
+        Put a the command 'next' field image scan on the queue with the appropriate field meta data model of the field
+        image to be scannend. Can only be executed if it proceeded by a 'start' mega field scan command on the queue.
+        As notifier function the dataflow.notify is given which means the returned image will be redirected to this
```

Comment:
the dataflow.notify is the notifier function to which output images (empty or not) will be redirected.

Reply:
Can you not just call self.data.notify() in the acquisition thread? I am missing why I need to provide the dataflow.notify() in the queue.

Reply:
For acquire_single_field we like to have the image not returned to the dataflow but to the queue inside this method so we can directly return the full image to the caller (in this case the dataflow.get() method) which sends the image not via a notify basis directly via a return. This is maybe a bit complicated construction but it is very flexible and neat, though for this implementation I need to share the credits with Eric.

Reply:
Ah ok. From the text I have trouble understanding this. Maybe this can be rephrased or a bit more detailed. However, I don't understand enough to suggest anything.
You don't return anything to this method here. What do you mean with "this function". The notifier function for the dataflow?

Reply:
Yes - easier to understand now!
Small suggestion, but you can also leave as is.
```
acq_thread -> acquisition thread
The acquired image will be returned to the dataflow via the notifier function, which ....
```